### PR TITLE
IWF-157: Fix typo IdReusePolicy of ALLOW_IF_PREVIOUS_EXISTS_ABNORMALLY

### PR DESCRIPTION
**Pull request by stevo89519**

### CODE REVIEW

**Review by samuel27m — APPROVED** [review]

**Comment by longquanzheng** [comment]
Can you look into why the tests failing

**Comment by stevo89519** [comment]
> Can you look into why the tests failing

@longquanzheng Fixed some linter warnings. Tests passed by themselves (no changes).
Linter is having dependency issues:
`Error: Command failed: poetry install`

**Comment by longquanzheng** [comment]
> > Can you look into why the tests failing
> 
> @longquanzheng Fixed some linter warnings. Tests passed by themselves (no changes). Linter is having dependency issues: `Error: Command failed: poetry install`

I guess we need to upgrade this to use official action: https://github.com/marketplace/actions/install-poetry-action

This is no longer working properly with old nodeJs: https://github.com/knowsuchagency/poetry-install/issues/4

**Comment by longquanzheng** [comment]
Could you also add a simple test like in Java ?

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -29,8 +29,8 @@ jobs:
         python-version: '3.9'
     - name: Install deps
       uses: snok/install-poetry@v1
-      env:
-        POETRY_VIRTUALENVS_CREATE: false
+      with:
+        virtualenvs-create: false
```
Comment (longquanzheng) [inline .github/workflows/tests.yml:33]: I copied this value from somewhere else but didn't fully understand. do you know why this has to be 'false'?

File: iwf/object_encoder.py [hunk 2]
```diff
@@ -291,7 +291,7 @@ def default(self, o: Any) -> Any:
         See :py:meth:`json.JSONEncoder.default`.
         """
         # Dataclass support
-        if dataclasses.is_dataclass(o):
+        if dataclasses.is_dataclass(o) and not isinstance(o, type):
```
Comment (stevo89519) [inline iwf/object_encoder.py:294]: Fixes mypy error

File: .github/workflows/tests.yml (hunk 1 above)
Comment (stevo89519) [reply .github/workflows/tests.yml:35]: No I don't know. I'm also not sure if calling 'poetry install' here will cause any downsides? It fixed the workflow issues.

File: iwf/object_encoder.py (hunk 2 above)
Comment (longquanzheng) [reply iwf/object_encoder.py:294]: I didn't see this error before. 
Did you find this fix somewhere? can you put a reference here for more context?

**Review by longquanzheng — APPROVED** [review]
Great job!

**Pull request merged**